pkg/reconciler/namespace/namespace.go

```diff
@@ -121,62 +126,80 @@ func (r *Reconciler) reconcile(ctx context.Context, ns *corev1.Namespace) error
 	if ns.DeletionTimestamp != nil {
 		return nil
 	}
-	if err := r.reconcileServiceAccountAndRoleBindings(ctx, ns, resources.IngressServiceAccountName, resources.IngressRoleBindingName, resources.IngressClusterRoleName, resources.ConfigClusterRoleName); err != nil {
-		return fmt.Errorf("broker ingress: %v", err)
+
+	cmp, err := r.reconcileConfigMapPropagation(ctx, ns)
+	if err != nil {
+		return fmt.Errorf("configMapPropagation: %w", err)
+	}
+
+	// Tell tracker to reconcile this namespace whenever the ConfigMapPropagation changes.
+	if err = r.tracker.Track(utils.ObjectRef(cmp, configMapPropagationGVK), ns); err != nil {
+		return fmt.Errorf("track configMapPropagation: %w", err)
 	}
 
-	if err := r.reconcileServiceAccountAndRoleBindings(ctx, ns, resources.FilterServiceAccountName, resources.FilterRoleBindingName, resources.FilterClusterRoleName, resources.ConfigClusterRoleName); err != nil {
-		return fmt.Errorf("broker filter: %v", err)
+	if err := r.reconcileServiceAccountAndRoleBindings(ctx, ns, resources.IngressServiceAccountName, resources.IngressRoleBindingName, resources.IngressClusterRoleName); err != nil {
+		return fmt.Errorf("broker ingress: %w", err)
+	}
+
+	if err := r.reconcileServiceAccountAndRoleBindings(ctx, ns, resources.FilterServiceAccountName, resources.FilterRoleBindingName, resources.FilterClusterRoleName); err != nil {
+		return fmt.Errorf("broker filter: %w", err)
 	}
 
 	b, err := r.reconcileBroker(ctx, ns)
 	if err != nil {
-		return fmt.Errorf("broker: %v", err)
+		return fmt.Errorf("broker: %w", err)
 	}
 
 	// Tell tracker to reconcile this namespace whenever the Broker changes.
 	if err = r.tracker.Track(utils.ObjectRef(b, brokerGVK), ns); err != nil {
-		return fmt.Errorf("track broker: %v", err)
+		return fmt.Errorf("track broker: %w", err)
 	}
 
 	return nil
 }
 
+// reconcileConfigMapPropagation reconciles the default ConfigMapPropagation for the Namespace 'ns'.
+func (r *Reconciler) reconcileConfigMapPropagation(ctx context.Context, ns *corev1.Namespace) (*configsv1.ConfigMapPropagation, error) {
+	current, err := r.EventingClientSet.ConfigsV1alpha1().ConfigMapPropagations(ns.Name).Get(resources.DefaultConfigMapPropagationName, metav1.GetOptions{})
+
+	// If the resource doesn't exist, we'll create it.
+	if k8serrors.IsNotFound(err) {
+		cmp := resources.MakeConfigMapPropagation(ns.Name)
+		cmp, err = r.EventingClientSet.ConfigsV1alpha1().ConfigMapPropagations(ns.Name).Create(cmp)
+		if err != nil {
+			return nil, err
+		}
+		r.Recorder.Event(ns, corev1.EventTypeNormal, configMapPropagationCreated,
+			"Default ConfigMapPropagation: "+cmp.Name+" created")
+		return cmp, nil
+	} else if err != nil {
```

Comment:
See previous comment about the use of `switch`.

Reply:
Same here, followed the same way as other methods in this go file did (like reconcileBroker(), reconcileBrokerRBAC())